### DISCUSSION TITLE
Rendering

### DISCUSSION THREAD

**Discussion by woutercouvaras** [codex-team/editor.js#1767] [Ideas]
Hi,

Firstly, I'm totally in love with editor js!  I love everything I've learned about it so far!

I use Nuxtjs (a Vue meta framework) which offers server side rendering (among many other things), and was able to get an working implementation with image storing pretty quickly and easily.  Winning! :)

The thing that I'm not sure about is, how does one/can one render the block content on the server?  I can't get it to work by default because it's looking for the window object.

Server rendering is pretty important from and SEO point of view, so I'm wondering if someone else has figured this out?

If this hasn't been solved yet, I thought it might be worth discussing some ideas around how this can be achieved.

One of the thoughts I had was that the `save` method could optionally return the block content and the rendered html, so that one could store both.  This may be a bit hacky, but I think it's possibly a solution that would require the least amount of work to get pre-rendered content.

Thoughts?

**Comment by Skywalker-72** [comment]
@woutercouvaras  You may get a few ideas here : https://github.com/angular/universal/issues/1675/

I have the same problem that you have mentioned but in Angular. I opened a new discussion on the topic and also there is a thread where me and @Teebo have tried a couple of solutions. You may wanna check that out as well
